### DISCUSSION TITLE
Questions about last update

### DISCUSSION THREAD

**Discussion by tizfra** [GeekFunkLabs/fluidpatcher#73] [Q&A] [answered]
Hi, 
I've found that I can't scroll up and down patches with midi cc since I've update my squishbox to 0.7.14.

I've also found that web manager write with wrong owner and permission when I try to copy a bank, permission is 0664 instead 0775 and owner is www-data:www-data.
What can I do?
Thanks

**Comment by ilragazzo** [comment]
I too am having problems scrolling through the patches after updating. 

I've been using this on and off for about a year and a half. And on my LaunchKey37 and RPi3 it seemed to work well enough for me, but when I started to try and really set it up in a better way it all stopped working. 

I've tried using the pots to select patches which used to work well. But now just shuts off the RPi. And I've tried to use pad buttons in an incremental way, (like how it's intended to be used), but nothing I do will make the patches or banks change. 

I've been rackin my brains for the last few days on how to troubleshoot it, & how to see what my midi controller is sending to the RPi. 

I'm not sure which files store the settings for what change controls change the patches or banks.

**Comment by albedozero** [comment, marked as answer]
Oops, there were two issues here. TL;DR you can re-run the install script (`curl -L git.io/squishbox | bash`) and it will work, except you'll need to edit [this line](https://github.com/albedozero/fluidpatcher/blob/63583c859f1d133136b46b7ae89479a71b25083b/headlesspi.py#L20) afterward if you want to use a knob/slider/pot to scroll through patches.

First, in a recent update I changed [these lines](https://github.com/albedozero/fluidpatcher/blob/c6314dfe8972b4b54a63dee11f1bb650aa848c0f/headlesspi.py#L16-L17) to [these](https://github.com/albedozero/fluidpatcher/blob/63583c859f1d133136b46b7ae89479a71b25083b/headlesspi.py#L17-L20) - I thought the SELECT_PATCH option was confusing to some people, and created surprising and odd behavior if they already had a control assigned to CC#24, so I set it to `None` by default. Second, I tried to add a universal `patch` router rule keyword to give people more control over using MIDI messages to select patches, but I messed up the syntax in _headlesspi.py_.

@tizfra the permissions and ownership are correct. The installer adds you to the www-data group, and permissions 664 means user+group can read/write, so you should be able to edit and move the banks. 775 would be user+group can read/write/execute.

Thanks for pointing out these issues - let me know if there are still problems!

**Reply by ilragazzo** [reply]
That resolved it. Yay I got my organ sound back!!! Thank you!
